lib/propolis/src/hw/pci/device.rs

```diff
@@ -608,10 +651,9 @@ impl DeviceState {
         let attach = inner.attached();
         for n in BarN::iter() {
             if let Some((def, addr)) = inner.bars.get(n) {
-                let pio_en = inner.reg_command.contains(RegCmd::IO_EN);
-                let mmio_en = inner.reg_command.contains(RegCmd::MMIO_EN);
-
-                if (pio_en && def.is_pio()) || (mmio_en && def.is_mmio()) {
+                if (inner.pio_en() && def.is_pio())
+                    || (inner.mmio_en() && def.is_mmio())
```

Comment:
nit, take it or leave it: I wonder if there ought to be a method like
```rust
impl State {
    fn should_register_bar(&self, bar: &BarDefine) -> bool {
         (self.pio_en() && bar.is_pio()) || (self.mmio_en() && def.is_mmio())
    }
}
```
that  could be called both here and in `cfg_std_write`, above?

Not sure if it's worth factoring out or not, just noticed that it was repeated.